cpp/src/link_analysis/pagerank.cu

```diff
@@ -1,286 +1,278 @@
-#else
-  ALLOC_FREE_TRY(d_pr, stream);
-#endif
-  ALLOC_FREE_TRY(d_leaf_vector, stream);
-
-}
-}
-
-template <typename VT, typename WT>
-void pagerank(Graph *graph,WT* pagerank, size_t personalization_subset_size,
-  VT* personalization_subset,WT* personalization_values,
-  float alpha, float tolerance, int max_iter, bool has_guess) {
-  //
-  //  Pagerank operates on CSR and can't currently support 64-bit integers.
-  //  If csr doesn't exist, create it.  Then check type to make sure it is 32-bit.
-  //
-  CUGRAPH_EXPECTS(graph->transposedAdjList != nullptr, "Invalid API parameter: Transposed Adj List is needed by pagerank");
-
-  if (typeid(WT) != typeid(float) && typeid(WT) != typeid(double))
-    CUGRAPH_FAIL("Unsupported weight data type, please use float or double");
-  
-  if (typeid(VT) != typeid(int) )
-    CUGRAPH_FAIL("Unsupported personalization_subset data type, please use int");
-
-  CUGRAPH_EXPECTS( pagerank != nullptr , "Invalid API parameter: Pagerank array should be of size V" );
-
-  if (personalization_subset_size != 0) {
-    CUGRAPH_EXPECTS(typeid(VT) == typeid(int), "Unsupported personalization_subset data type");
-  }
-
-  return detail::pagerank_impl<VT,WT>(graph, pagerank,
-                                  personalization_subset_size, personalization_subset, personalization_values,
-                                  alpha, tolerance, max_iter, has_guess);
-}
-
-// explicit instantiation
-template void pagerank<int, float>(Graph *graph,float* pagerank,
-  size_t personalization_subset_size, int* personalization_subset,float* personalization_values,
-  float alpha, float tolerance, int max_iter, bool has_guess);
-template void pagerank<int, double>(Graph *graph,double* pagerank,
-  size_t personalization_subset_size, int* personalization_subset,double* personalization_values,
-  float alpha, float tolerance, int max_iter, bool has_guess);
-
-} //namespace cugraph 
\ No newline at end of file
+/*
+ * Copyright (c) 2018-2020, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * NVIDIA CORPORATION and its licensors retain all intellectual property
+ * and proprietary rights in and to this software, related documentation
+ * and any modifications thereto.  Any use, reproduction, disclosure or
+ * distribution of this software and related documentation without an express
+ * license agreement from NVIDIA CORPORATION is strictly prohibited.
+ *
+ */
+
+// Pagerank solver
+// Author: Alex Fender afender@nvidia.com
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stddef.h>
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <iomanip>
+#include "cub/cub.cuh"
+#include <algorithm>
+#include <iomanip>
+
+#include <rmm_utils.h>
+
+#include "utilities/graph_utils.cuh"
+#include "utilities/error_utils.h"
+#include <cugraph.h>
+#include <graph.hpp>
+
+namespace cugraph { 
+namespace detail {
+
+#ifdef DEBUG
+  #define PR_VERBOSE
+#endif
+
+template <typename IndexType, typename ValueType>
```

Comment:
I think we should replace IndexType with VT and ET to be future proof.

Reply:
I left these because the current implementation assumes both VT and ET are int.  Changing to support VT and ET being different sizes and supporting int64_t for either of these types would require more extensive changes, I kind of felt like that was appropriate for a different PR.

I felt like partially changing this would be deceptive and perhaps might make us think that this would work for the future data structure.

Reply:
I see, then could you submit another issue report on this problem, so we'll not forget about this?

Reply:
Created https://github.com/rapidsai/cugraph/issues/729